### DISCUSSION TITLE
The problem of playing videos on iPhone devices (Load videos)

### DISCUSSION THREAD

**Discussion by avcr2019** [airsdk/Adobe-Runtime-Support#2650] [General]
Welcome all my friends
I wish you all a happy day
I have a problem and I don't know how to solve it after several trials
I have an application running on the Android system and successfully. And the videos are summoned and played, and if they are not present on the user’s device, they are downloaded from a server, and the video is downloaded and works.
The problem here is that I want to export the app to Apple devices but the code itself is not working. Since he does not want to recall the video, even if it is already in the files. It comes with a white screen and the video is not called. Here is the code used to call the video, which works successfully on Android, but on iPhone it does not work at all

`
import flash.net.NetConnection;
import flash.net.NetStream;
import flash.media.Video;
import flash.filesystem.File;
import flash.media.Sound;
import flash.net.URLRequest;

var connection:NetConnection;
var mt:Object;
var stream:NetStream;
var video:Video;

function loadVideo():void{
    connection = new NetConnection();
	connection.connect(null);

	mt = new Object();
	mt.onMetaData = onMeta;
	function onMeta(obj:Object):void{};

	stream = new NetStream (connection);
	stream.client = mt;
	if(isLocal){
		stream.play ("videos/" + Config.currentVideo);
	}else{
		stream.play (file2.url);
	}
	

	video = new Video();
	video.attachNetStream(stream);
	 stage.addChild(video);

	video.width = stage.stageWidth ;
	video.height = stage.stageHeight ;

video.width = 1175;
video.height = 624;
video.x = 56;
video.y = 44;

	play_mc2.addEventListener(MouseEvent.CLICK, playvid);
	pause_mc2.addEventListener(MouseEvent.CLICK, pausevid);
	stop_mc2.addEventListener(MouseEvent.CLICK, stopvid);

}

function playvid(e:MouseEvent):void
{
	stream.resume();
}

function pausevid(e:MouseEvent):void
{
	stream.pause();
}

function stopvid(e:MouseEvent):void
{
	stream.pause();
	stream.seek(0);
}

var file:File = File.applicationDirectory.resolvePath("videos"+ File.separator + Config.currentVideo);
var file2:File = File.applicationStorageDirectory.resolvePath("media"+File.separator+"videos"+File.separator+Config.currentVideo);
`
Will the video export settings be changed?
Or where is the problem, especially since I used another code, and the video is not called up either
`import flash.net.NetConnection;
import flash.net.NetStream;
import flash.media.Video;
import flash.events.*;
var connectionintro1:NetConnection = new NetConnection();
connectionintro1.connect(null);
var mtintro1:Object = new Object();
mtintro1.onMetaData = onMetaintro1;
function onMetaintro1(objintro1:Object):void{};
var streamintro1:NetStream = new NetStream (connectionintro1);
streamintro1.addEventListener(NetStatusEvent.NET_STATUS, netStatusHandlerintro1);
streamintro1.client = mtintro1;
streamintro1.play ("videos/test.mp4");
var videointro1:Video = new Video();
videointro1.attachNetStream(streamintro1);
stage.addChild(videointro1);
videointro1.width = stage.stageWidth;
videointro1.height = stage.stageHeight ;
function netStatusHandlerintro1(event:NetStatusEvent):void {
if (event.info.code == "NetStream.Play.Stop") {
stage.removeChild(videointro1);
streamintro1.close();
gotoAndPlay(1, "main");
}
}
`

**Comment by itlancer** [comment]
For iOS `Video` object doesn't playback MP4 H.264 videos. Here bug report: https://github.com/airsdk/Adobe-Runtime-Support/issues/211
Use `StageVideo` or `Stage3D` `VideoTexture` as workaround.

**Reply by avcr2019** [reply]
> For iOS `Video` object doesn't playback MP4 H.264 videos. Here bug report: #211 Use `StageVideo` or `Stage3D` `VideoTexture` as workaround.

Thank you for the reply, my friend
Well, he went to the link that I sent and read it all, and I tried, but every time I failed to call the video successfully, but I tried converting the video from (MP4 H.264) to (FLV H.264) and indeed the video was called and it worked successfully, but another problem occurred, which is that when going to any other video. It does not work and shows a white screen only, but the sound works in the background, and this means that the video has been called, but it does not work correctly. I want you to help me solve this problem, and is there a project in the form of (FLA) and there is any of these (StageVideo or Stage3D VideoTexture)

**Reply by itlancer** [reply]
@avcr2019, using FLV with **H.263** (not H.264) codec works fine for me with iOS using `Video` object.
Make sure you correctly dispose and clear all resources before start new one video.
Take a look at sample here https://github.com/airsdk/Adobe-Runtime-Support/issues/2402
This part I talking about (previous video resources disposing):
```
private function startVideo():void {
	removeChildren();
	
	if (ns != null){
		video.attachNetStream(null);
		ns.removeEventListener(NetStatusEvent.NET_STATUS, nsHandler);
		ns.dispose();
	}

	if (video != null){
		video.clear();
	}
	
	video = new Video(640, 480);
	addChild(video);
	
	if (nc != null){
		nc.removeEventListener(NetStatusEvent.NET_STATUS, ncHandler);
		nc.close();
	}
	
	nc = new NetConnection();
	nc.addEventListener(NetStatusEvent.NET_STATUS, ncHandler);
	nc.connect(null);
}
```

`StageVideo` usage sample you can download from here:
https://github.com/airsdk/Adobe-Runtime-Support/issues/1942

`VideoTexture` usage sample you can download from here:
https://github.com/airsdk/Adobe-Runtime-Support/issues/2296
but simplier to use Starling or Feathers for GPU-accelerated content:
https://forum.starling-framework.org/d/14837-video-player/39
https://feathersui.com/learn/as3-starling/video-player/

**Reply by avcr2019** [reply]
Thank you bro
